### ISSUE TITLE
Document Raspi Zero W Setup for Arch Linux ARM

### ISSUE THREAD

**Issue by truedat101** [RazortoothRTC/raspizero-archlinux-base#1]
Use this issue to document the Raspi Zero W set up with Arch Linux ARM

**Comment by truedat101** [comment]
General notes and links on wpa supplicant / linux: 

https://wiki.netbsd.org/tutorials/how_to_use_wpa_supplicant/ 

https://wiki.archlinux.org/index.php/WPA_supplicant 

https://donnutcompute.wordpress.com/2014/04/20/connect-to-wi-fi-via-command-line/ 

https://linuxconfig.org/setup-wireless-interface-with-wpa-and-wpa2-on-ubuntu 

https://stackoverflow.com/questions/12899460/how-to-debug-wpa-supplicant-in-android 

wpa_supplicant -B -iwlan0 -c/etc/wpa_supplicant.conf -d

**Comment by existere** [comment]
Installed VirtualBox on macOS and Debian on VirtualBox VM. 
Created partitions on SC card for Arch Linux install. 
Downloaded Arch Linux and moved boot files to partition. 
Connected RPi but no signal, black screen.
Downloading Arch Linux again to reformat sd card to try again.

**Comment by existere** [comment]
Arch Linux will not boot into RPi, no power lights activated. Not sure what the source of the problem is, but I will try installing NOOBS next to make sure I have a working PRi.

**Comment by truedat101** [comment]
can you create the SD card from os x?

**Comment by existere** [comment]
I can try that. I was using virtualbox so that I could use Debian but it probably just complicated the process unnecessarily. However, the SD card seems properly formatted when reading it from the last step of the install in Debian before attempting to boot into the RPi.

**Comment by truedat101** [comment]
a bad SD card won't boot.  you will end up with the colors on the
bootloader screen .

**Comment by existere** [comment]
I'm not getting any colors or indicator lights. I assume the SD is the issue.

**Comment by existere** [comment]
The instructions on the arch linux site are for linux, which is also why I chose to use debian with vm. I'll need to find a set of instructions for mac.

**Comment by existere** [comment]
https://raspberrypi.stackexchange.com/questions/24118/install-arch-linux-from-osx

**Comment by truedat101** [comment]
so the question is which arch Linux distro did you start with?

**Comment by existere** [comment]
The ARM latest version, which I got from their website: https://archlinuxarm.org/platforms/armv6/raspberry-pi

**Comment by truedat101** [comment]
Ok, I guess start with noobs.  



But we need to revisit the card building for arch.

**Comment by truedat101** [comment]
In my experience, the problems with the boot / sdcard are typically an error in the set up.  Since it isn't automated, you have to do a perfect input through all the steps.  What I suggest is to verify the card after you create it in gparted (post a picture of the card layout ) .

**Comment by existere** [comment]
![Screen Shot 2020-09-15 at 11 32 51 AM](https://user-images.githubusercontent.com/58608271/93238617-4c1e1500-f747-11ea-87cf-ae38b60c9f9b.png)

I rewrote the image to the sd card. For some reason the file system on the first partition is saying it is fat16 even though I followed the command for fat32.

**Comment by existere** [comment]
I just connected the rpi again and now I have ACT light on but the screen is still blank with no arch boot.

**Comment by truedat101** [comment]
I think try noobs to just make sure this thing is working at all, and if
you have a second card use that.  One thing:

you must sync after you untar the root and boot partitions onto the card
because data may still be written for up to 5 minutes.

And unmounting the card it is critical that you unmount / eject the sdcard
via the OS so all resources are freed cleanly.  Still, let's make sure this
thing boots into something first.

**Comment by existere** [comment]
Ok, I made sure to sync after the untar _and_ after the boot _and_ made sure to unmount correctly and now it's booting into arch!

**Comment by existere** [comment]
arch booted
wifi enabled
pacman initialized 
installing desktop

**Comment by existere** [comment]
Installation broke at desktop manager install (sddm). Looking for another option as I believe that option is no longer compatible with rpi.

**Comment by existere** [comment]
Rebooted, now stuck on Load/Save RF Kill Switch Status. Not sure how to get past this.

**Comment by truedat101** [comment]
Can you post a screenshot of the problem?

**Comment by existere** [comment]
Finally figured out that the rpi reboot stops there normally and you have to use alt+f2 to toggle to arch. Working on getting the desktop set up, arch is up and running.

**Comment by truedat101** [comment]
Great.  If you don't mind to document any additional steps in to set up the Desktop, and any particular issues you ran into.

I advise the most minimal Desktop Set up as the 1 core is not a strong work horse.

https://github.com/RazortoothRTC/raspizero-archlinux-base/wiki/ArchLinux-ARM-Raspi0-W-Desktop-Setup-Guide

**Comment by existere** [comment]
When I was attempting to set it up earlier I ran into some issues trying to start the GUI following this guide:

[https://www.instructables.com/id/Arch-Linux-on-Raspberry-Pi/](url)
and
[https://www.zybuluo.com/yangxuan/note/344907](url)

It would not start when I ran `sudo systemctl start sddm` and I'm getting `Could not resolve host: mirror.archlinuxarm.org` when attempting to download some packages. Also I couldn't get Packer to work although it is installed.  

I'm reading about a recent change in Arch with something called systemD that relies on a specific configuration of DNS and NTP to access the packages necessary for the desktop. Curious if you've heard of anything like this. Tomorrow afternoon I will try and find some more information on it and try to reinstall everything for another go. 

From a comment on the first guide: 

"...you will notice that the DNS is not working hmmm why? it is because systemd does not care about /etc/resolv.conf it has hardcoded DNS servers such as 1.1.1.1 which is some DNS server from australia... anyway still the DNS servers are online so they should work so what is going on? What is going on is that DNS sec is enabled by default and it relies on computer time but your time is way off thanks to systemd and NTP is not installed."

Related post: 
[https://www.reddit.com/r/linuxquestions/comments/cf6my0/raspberry_pi_3_with_archlinuxarm_cant_update/](url)
